cookbooks/bcpc-hadoop/recipes/maven_config.rb

```diff
@@ -2,15 +2,89 @@
 
 node.override['maven']['install_java'] = false
 
+if ::Dir.exist?('/usr/local/share/ca-certificates')
+  node.override['maven']['mavenrc']['opts'] = \
+    node['maven']['mavenrc']['opts'] +\
+    " -Djavax.net.ssl.trustStorePassword=doesnotneedtobesecure " + \
+    "-Djavax.net.ssl.trustStore=#{node['bcpc']['hadoop']['java_https_keystore']}"
+end
+
 internet_download_url = node['maven']['url']
 maven_file = Pathname.new(node['maven']['url']).basename
 
 node.override['maven']['url'] = "#{get_binary_server_url}/#{maven_file}"
 
+unified_certs = ::File.join(Chef::Config[:file_cache_path], 'ca-certs.pem')
+
 # download Maven only if not already stashed in the bins directory
 remote_file "/home/vagrant/chef-bcpc/bins/#{maven_file}" do
   source internet_download_url
   action :create
   mode 0555
   checksum node['maven']['checksum']
 end
+
+include_recipe 'maven::default'
+
+file 'cacert file' do
+  cert_data = Dir.glob('/usr/local/share/ca-certificates/*').map do |cert|
+    File.open(cert, 'r').read()
+  end.join("\n")
+  path unified_certs 
+  content lazy { cert_data }
+  action :create
+  only_if { ::Dir.exist?('/usr/local/share/ca-certificates') }
+  notifies :delete, 'file[keystore file]', :immediately
+  notifies :run, 'execute[create keystore]', :immediately
+end
+
+file 'keystore file' do
+  path node['bcpc']['hadoop']['java_https_keystore']
+  action :nothing
+end
+
+execute 'create keystore' do
+  command <<-EOH
+    yes | keytool -v -alias mavensrv -import \
+    -file #{unified_certs} \
+    -keystore #{node['bcpc']['hadoop']['java_https_keystore']} \
+    -storepass doesnotneedtobesecure \
+    -alias mavensrv -import
```

Comment:
seems dangerous to configure a global java SSL thing in a recipe called "maven_config"